src/components/Drawer.js

```diff
@@ -238,92 +238,97 @@ class Drawer extends React.Component {
+            const titleUniqueId = _uniqueId('mx-drawer-title-');
+
+            return (
+              <div className='mx-drawer' onKeyUp={typeof this.props.onKeyUp === 'function' ? this.props.onKeyUp : this._handleKeyUp} style={styles.componentWrapper}>
+                <div
+                  className='mx-drawer-scrim'
+                  onClick={() => {
+                    if (this.props.closeOnScrimClick) this.close();
+                  }}
+                  style={styles.scrim}
+                />
+                <div
+                  aria-describedby={this.props['aria-describedby']}
+                  aria-labelledby={this.props['aria-labelledby'] || titleUniqueId}
+                  ref={(ref) => (this._component = ref)}
+                  role={this.props.role}
+                  style={{ ...styles.component, ...this.props.style }}
+                  tabIndex={0}
+                >
+                  <header className='mx-drawer-header' style={{ ...styles.header, ...this.props.headerStyle }}>
+                    <span style={styles.backArrow}>
+                      {this.props.showCloseButton &&
+                        <Button
+                          aria-label={closeButtonAriaLabel || `Close ${this.props.title} Drawer`}
+                          buttonRef={ref => (this._closeButton = ref)}
+                          className='mx-drawer-close'
+                          icon='go-back'
+                          onClick={this.close}
+                          theme={theme}
+                          type={'base'}
+                        />
+                      }
+                    </span>
+                    <h2 id={titleUniqueId} style={styles.title}>
+                      {this.props.title}
+                    </h2>
+                    <div className='mx-drawer-header-menu' style={styles.headerMenu}>
+                      {menu}
+                    </div>
+                  </header>
+                  <div className='mx-drawer-content' style={{ ...styles.content, ...this.props.contentStyle }}>
+                    {typeof this.props.children === 'function' ? this.props.children(this._getExposedDrawerFunctions()) : this.props.children}
+                  </div>
                 </div>
-              </header>
-              <div className='mx-drawer-content' style={{ ...styles.content, ...this.props.contentStyle }}>
-                {typeof this.props.children === 'function' ? this.props.children(this._getExposedDrawerFunctions()) : this.props.children}
               </div>
-            </div>
-          </div>
+            );
+          }}
         </MXFocusTrap>
       </ StyleRoot>
     );
   }
 
-  styles = (theme) => {
+  styles = (theme, trapNumber = 1) => {
     const HEADER_HEIGHT = this._getHeaderHeight();
 
     return _merge({}, {
       component: {
         border: '1px solid ' + theme.Colors.GRAY_300,
         boxSizing: 'border-box',
-        zIndex: 1001,
+        zIndex: trapNumber * 1002,
```

Comment:
I am not sure multiplication is what we want here. Do you want it to increment by the thousands for each number?

Reply:
If I do addition then I have to know the zIndex of the first trap rendered to make sure we don't have overlap with numbers.  This is due to the fact that the wrapping div, scrim, and content all have individual z index values because of how they are positioned.

Multiplication solves that problem as the zIndex value will always be unique.

Reply:
I am not sure I understand how multiplication ensures the zIndex value is always unique. Can you give me an example of the difference?